tests/platforms/linux/system/test__find_changelog_filename.py

```diff
@@ -0,0 +1,48 @@
+import pytest
+
+from briefcase.platforms.linux.system import find_changelog_filename
+
+from ....utils import create_file
+
+
+def test_no_changelog(tmp_path):
+    assert find_changelog_filename(tmp_path) is None
+
+
+def test_no_valid_changelog(tmp_path):
+    """Changelog file exists but with invalid name format so its not accepted"""
+    changelog_file = tmp_path / "base_path/invalid_changelog_format"
+    create_file(changelog_file, "First App Changelog")
+    assert find_changelog_filename(tmp_path / "base_path") is None
+
+
+@pytest.mark.parametrize(
+    "changelog_filename",
+    [
+        f"{name}{extension}"
+        for name in ["CHANGELOG", "HISTORY", "NEWS", "RELEASES"]
+        for extension in ["", ".md", ".rst", ".txt"]
+    ],
+)
+def test_has_changelog(tmp_path, changelog_filename):
+    """Makes sure all formats are accepted"""
+    changelog_file = tmp_path / f"base_path/{changelog_filename}"
+    create_file(changelog_file, "First App Changelog")
+    assert find_changelog_filename(tmp_path / "base_path") == changelog_filename
+
+
+def test_multiple_changefile(tmp_path):
+    """
+    More than one changelog exists but only first one according
+    the below search order is accepted:
+
+    CHANGELOG, CHANGELOG.md, CHANGELOG.rst, CHANGELOG.txt,
+    HISTORY, HISTORY.md, HISTORY.rst, HISTORY.txt,
+    NEWS, NEWS.md, NEWS.rst, NEWS.txt,
+    RELEASES, RELEASES.md, RELEASES.rst, RELEASES.txt"""

```

Comment:
There's no need to enumerate the full logic here. 
```suggestion
    """If there's more than one changelog, only one if found."""
```